spacy/ml/tb_framework.pyx

```diff
@@ -43,18 +45,28 @@ def TransitionModel(
     tok2vec_projected = chain(tok2vec, list2array(), Linear(hidden_width, t2v_width))  # type: ignore
     tok2vec_projected.set_dim("nO", hidden_width)
 
+    # Fixme: we use `upper` as a container for the upper layer's
+    # weights and biases. Thinc optimizers cannot handle resizing
+    # of parameters. So, when the parser model is resized, we
+    # construct a new `upper` layer, which has a different key in
+    # the optimizer. Once the optimizer supports parameter resizing,
+    # we can replace the `upper` layer by `upper_W` and `upper_b`
+    # parameters in this model.
+    upper = Linear(nO=None, nI=hidden_width, init_W=zero_init)
```

Comment:
Would it more preferable to rename `upper` to `output` here? Using a positional name that implicitly depends on how one visualises the topology of the layers can be a bit confounding, I think (on first glance, I implicitly associated `upper` with `input_layer` as I tend to visualise a network in a top-down manner).